akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowInterleaveSpec.scala

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl
+
+import akka.stream.testkit.Utils._
+import akka.stream.testkit._
+import org.reactivestreams.Publisher
+
+class FlowInterleaveSpec extends BaseTwoStreamsSetup {
+
+  override type Outputs = Int
+
+  override def setup(p1: Publisher[Int], p2: Publisher[Int]) = {
+    val subscriber = TestSubscriber.probe[Outputs]()
+    Source(p1).interleave(Source(p2), 2).runWith(Sink(subscriber))
+    subscriber
+  }
+
+  "An Interleave for Flow " must {
+
+    "work in the happy case" in assertAllStagesStopped {
+      val probe = TestSubscriber.manualProbe[Int]()
+      Source(0 to 3).interleave(Source(List[Int]()), 2).interleave(Source(4 to 9), 3).runWith(Sink(probe))
+
+      val subscription = probe.expectSubscription()
+
+      var collected = Set.empty[Int]
```

Comment:
since this merge shall be deterministic we shouldn’t need Sets, but Seq should work, right?